lib/Basics/expected.h

```diff
@@ -201,16 +201,19 @@ struct expected {
 template<>
 struct expected<void> {
   expected() = default;
-  explicit expected(std::exception_ptr ex) : _exception(std::move(ex)) {}
-  void reset() { _exception = nullptr; }
+  explicit expected(std::exception_ptr ex) noexcept
+      : _exception(std::move(ex)) {}
```

Comment:
it is unspecified whether `std::exception_ptr` is nothrow move constructible or not. Either add a static assert or make the `noexcept` conditional.

Reply:
We had the static assert, I moved it in one commit to the top of the file, but moved it back here to make it clear why it is required.